File: modules/swagger-parser/src/test/scala/SwaggerParserExtensionTest.scala
import com.wordnik.swagger.util.Json
import io.swagger.parser._
import io.swagger.parser.{Swagger20Parser, SwaggerParser}

import org.junit.runner.RunWith
import org.scalatest.junit.JUnitRunner
import org.scalatest.FlatSpec
import org.scalatest.Matchers

@RunWith(classOf[JUnitRunner])
class SwaggerParserExtensionTest extends FlatSpec with Matchers {
  val m = Json.mapper()

  it should "verify the default parser extension" in {
    val parser = new SwaggerParser()
    val extensions = parser.getExtensions()
    extensions.size should be > (0)
    extensions.get(0).getClass should be (classOf[Swagger20Parser])
  }
}

File: modules/swagger-parser/src/test/java/io/swagger/parser/SwaggerParserExtensionTest.java
package io.swagger.parser;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertTrue;

import org.testng.annotations.Test;

import java.util.List;

public class SwaggerParserExtensionTest {
    @Test
    public void verifyDefaultExtension() {
        final SwaggerParser parser = new SwaggerParser();
        final List<SwaggerParserExtension> extensions = parser.getExtensions();
        assertTrue(extensions.size() > 0);
        assertTrue(extensions.get(0) instanceof Swagger20Parser);
        assertEquals(extensions.get(0).getClass(), Swagger20Parser.class);
    }
}
